lib/problem_check/group_email_credentials.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+##
+# If group SMTP or IMAP has been configured, we want to make sure the
+# credentials are always valid otherwise emails will not be sending out
+# from group inboxes. This check is run as part of scheduled admin
+# problem checks, and if any credentials have issues they will show up on
+# the admin dashboard as a high priority issue.
+class ProblemCheck::GroupEmailCredentials < ProblemCheck
+  self.perform_every = 30.minutes
+
+  def call
+    [*smtp_errors, *imap_errors]
+  end
+
+  private
+
+  def smtp_errors
+    return [] if !SiteSetting.enable_smtp
+
+    Group.with_smtp_configured.find_each.filter_map do |group|
+      try_validate(group) do
+        EmailSettingsValidator.validate_smtp(
+          host: group.smtp_server,
+          port: group.smtp_port,
+          username: group.email_username,
+          password: group.email_password,
+        )
+      end
+    end
+  end
+
+  def imap_errors
+    return [] if !SiteSetting.enable_imap
+
+    Group.with_imap_configured.find_each.filter_map do |group|
+      try_validate(group) do
+        EmailSettingsValidator.validate_imap(
+          host: group.smtp_server,
+          port: group.smtp_port,
+          username: group.email_username,
+          password: group.email_password,
+        )
+      end
+    end
+  end
+
+  def try_validate(group, &blk)
+    begin
+      blk.call
+      nil
+    rescue *EmailSettingsExceptionHandler::EXPECTED_EXCEPTIONS => err
+      message =
+        I18n.t(
+          "dashboard.group_email_credentials_warning",
+          {
+            base_path: Discourse.base_path,
+            group_name: group.name,
+            group_full_name: group.full_name,
+            error: EmailSettingsExceptionHandler.friendly_exception_message(err, group.smtp_server),
+          },
+        )
+
+      Problem.new(message, priority: "high", identifier: "group_#{group.id}_email_credentials")
```

Comment:
I am also envisioning that this `Problem` class will be the basis of the new messaging system for admin, but perhaps `Problem` would just be one type of these messages.

Thinking of a generic class name, I think maybe `AdminAlert` might be nice (cool alliteration too), or maybe `AdminNotice` or something else. So in this case you would make `AdminAlert.new(message, type: "problem", priority: "high", identifier: "group_#{group.id}_email_credentials")` for example.

Some other options might be:

* dismissable: `Boolean` - Whether the alert can be dismissed by the user
* dismiss_for: `ActiveSupport::Duration` (default 6 months) - The amount of time that the Alert will be dismissed for if dismissable is true
* actions: `Array[AdminAlert::Action]` - Zero or more actions that can automatically be taken to deal with this alert. An action might for example have a label, an icon, and then a Proc or maybe a Job to run when clicked
* additional_alert: `AdminAlert::Action` - An additional action that automatically happens when this Alert is surfaced on the dashboard, e.g. sending a PM to admin. This will probably be a bit complex

Hope this makes sense, nothing actionable really I am just going off on a jazz riff :saxophone:

Reply:
This sounds like a very good extension to me. 👍